### ISSUE TITLE
Autogenerated IDL artifacts in the wrong directory

### ISSUE THREAD

**Issue by KiriakosD** [objectcomputing/OpenDDS#1630]
We are trying to build OpenDDS in our application and although everything is going well so far, one of the problems we are facing is that the autogenerated IDL artifacts are created in the root directory (where we have our CMakeLists.txt).

Is it possible to configure cmake to add them in a different directory?
Currently we are building our topics like this:

```
OPENDDS_TARGET_SOURCES(debug
    src/interface/topics/Messenger.idl
)
```
What we would like ideally would be to have the generated files under a src/interface/topics/messenger/ directory.

**Comment by KiriakosD** [comment]
Ok, I think it works if I add:
`OPENDDS_IDL_OPTIONS -o src/interface/topics/messenger`

**Comment by KiriakosD** [comment]
Also adding:
`TAO_IDL_OPTIONS -o src/interface/topics/messenger`

**Comment by iguessthislldo** [comment]
I'm not clear why you want to manually specify the destination of the IDL generation, but I wouldn't recommend doing that at all for CMake, as that would mess up the build. If you are getting stuff in the same directory as your `CMakeLists.txt`, then it sounds like you're just running `cmake` in place.  Having it work "in-source" will work, but ideally you'd run `cmake` from the directory you want to act as the "build" directory and pass it the path to the directory with `CMakeLists.txt`, like we do in [our example](https://github.com/objectcomputing/OpenDDS/blob/master/docs/cmake.md#unix).  See [`CMAKE_CURRENT_BINARY_DIR`](https://cmake.org/cmake/help/latest/variable/CMAKE_CURRENT_BINARY_DIR.html) and the concept of "out-of-source" builds in general for more information.

**Comment by iguessthislldo** [comment]
I tried it out and arbitrarily setting the output of the idl compilers does build with `-o` options, which is not what I expected. I still wouldn't recommend it though.

**Comment by iguessthislldo** [comment]
I also wanted to add that I wouldn't recommend relying on it because we need to fix #1297 still and fixing that might involve managing the generated IDL files in a different way.

**Comment by KiriakosD** [comment]
Thanks for your reply @iguessthislldo 
I am not sure I fully understand what can go wrong, but I will try to change it.
The reason why I wanted to do it, was because the output of the idl compilers was created in the directory of the source code, and that made things a bit "untidy".

I will start by looking into `CMAKE_CURRENT_BINARY_DIR`
